crates/bevy_reflect/src/lib.rs

```diff
@@ -1927,6 +1927,73 @@ bevy_reflect::tests::Test {
         );
     }
 
+    #[test]
+    fn dynamic_types_debug_format() {
+        #[derive(Debug, Reflect)]
+        struct TestTupleStruct(u32);
+
+        #[derive(Debug, Reflect)]
+        enum TestEnum {
+            A(u32),
+            B,
+        }
+
+        #[derive(Debug, Reflect)]
+        // test DynamicStruct
+        struct TestStruct {
+            // test DynamicTuple
+            tuple: (u32, u32),
+            // test DynamicTupleStruct
+            tuple_struct: TestTupleStruct,
+            // test DynamicList
+            list: Vec<u32>,
+            // test DynamicArray
+            array: [u32; 3],
+            // test DynamicEnum
+            e: TestEnum,
+            // test DynamicMap
+            map: HashMap<u32, u32>,
+            // test reflected value
+            value: u32,
+        }
+        let mut map = HashMap::new();
+        map.insert(9, 10);
+        let mut test_struct = TestStruct {
+            tuple: (0, 1),
+            list: vec![2, 3, 4],
+            array: [5, 6, 7],
+            tuple_struct: TestTupleStruct(8),
+            e: TestEnum::A(11),
+            map,
+            value: 12,
+        }
+        .clone_value();
+        let test_struct = test_struct.downcast_mut::<DynamicStruct>().unwrap();
+
+        // test unknown DynamicStruct
+        let mut test_unknown_struct = DynamicStruct::default();
+        test_unknown_struct.insert("a", 13);
+        test_struct.insert("unknown_struct", test_unknown_struct);
+        // test unknown DynamicTupleStruct
+        let mut test_unknown_tuple_struct = DynamicTupleStruct::default();
+        test_unknown_tuple_struct.insert(14);
+        test_struct.insert("unknown_tuplestruct", test_unknown_tuple_struct);
+        assert_eq!(
+            format!("{:?}", test_struct),
+            "DynamicStruct(bevy_reflect::tests::TestStruct { \
+                tuple: DynamicTuple((0, 1)), \
+                tuple_struct: DynamicTupleStruct(bevy_reflect::tests::TestTupleStruct(8)), \
+                list: DynamicList([2, 3, 4]), \
+                array: DynamicArray([5, 6, 7]), \
+                e: DynamicEnum(A(11)), \
+                map: DynamicMap({9: 10}), \
+                value: 12, \
+                unknown_struct: DynamicStruct( { a: 13 }), \
```

Comment:
```rs
unknown_struct: DynamicStruct( { a: 13 }), \
//                       -----^
```
Can you remove that space? Gonna require a bit more work than `.unwrap_or("")` I guess.

Reply:
I think this is just a limitation of `debug_struct`. We could manually format everything, but I’m okay with a tiny space— especially if it's not super likely to appear in debug output.

Reply:
I came to the same conclusion, it didn't seem worth the added complexity.

Reply:
would using `debug_map` instead of `debug_struct` for that particular use-case be too complex?

Reply:
Ah okay then, so they have different formatting. Sorry for being too lazy to check.

If we *have* to have a name, I'd prefer `DynamicStruct(_ { a: 13 })` or `DynamicStruct(? { a: 13 })`, because asymmetric space just looks like a formatting bug.

Reply:
I'm really not fond of `?`, but `_` is fine. Although it brings the question: do `DynamicTupleStruct`s also include it?
It would look like this: `DynamicTupleStruct(_(14))` vs. `DynamicTupleStruct((14,))`.
I think it should, to be more consistent.

Reply:
I had tried actually, but `PadAdapter` is not public API, and uses `Formatter::wrap_buf()` which is private too. I'm not sure it's possible at all, even if we wanted to copy lots of code.

Reply:
I went with _ for both structs and tuple structs, it seems to be the most correct of the discussed solutions.